viewer/packages/tools/src/SmartOverlay/SmartOverlayTool.ts

```diff
@@ -372,22 +284,17 @@ export class SmartOverlayTool extends Cognite3DViewerToolBase {
     intersection = intersection.filter(([icon, _]) => icon.intersect(_raycaster.ray) !== null);
 
     intersection
-      .map(([icon, intersection]) => [icon, intersection.sub(_raycaster.ray.origin)] as [Overlay3DIcon, THREE.Vector3])
+      .map(
+        ([icon, intersection]) =>
+          [icon, intersection.sub(_raycaster.ray.origin)] as [Overlay3DIcon<MetadataType>, THREE.Vector3]
+      )
       .filter(([, intersection]) => intersection.dot(cameraDirection) > 0)
       .sort((a, b) => a[1].length() - b[1].length());
```

Comment:
If we're only interested in one intersection, I think it's faster to go through the list to find the closest, instead of sorting all the points. Even though this is prettier :P

Reply:
Good point.